xray/conventions.py

```diff
@@ -826,10 +826,10 @@ def stackable(dim):
         if decode_coords:
             var_attrs = new_vars[k].attrs
             if 'coordinates' in var_attrs:
-                var_coord_names = var_attrs['coordinates'].split()
+                coords = pop_to(var_attrs, new_vars[k].encoding, 'coordinates')
+                var_coord_names = coords.split()
                 if all(k in variables for k in var_coord_names):
                     coord_names.update(var_coord_names)
-                    del var_attrs['coordinates']
```

Comment:
Yes, exactly. WRF, for example, produces netCDF files with a `coordinates` attribute that refers to variables that don't exist in the file.